src/GitVersionCore/MergeMessage.cs

```diff
@@ -72,17 +76,7 @@ private SemanticVersion ParseVersion(string branchName, string tagPrefix)
             return null;
         }
 
-        private class MergeMessagePattern
```

Comment:
I'm sorry for not noticing this before, but why have you removed the `MergeMessagePattern` class?

Reply:
I'd introduced the private class in the refactoring and unit testing completed in #1487. The class was unnecessary as it was simple a keyvalue pair so I switched out for that instead.

Reply:
I quite like having utility classes like that. Makes the code much more readable. Can you please bring it back? :pray: :smiley:

Reply:
Done.  This highlighted that there were a few different words being used inconsistently (Definition, Format, Pattern).  I've tidied that up too:
Each entry from the config and defaults creates a `MergeMessageFormat` instance.  A `MergeMessageFormat` consists of a `Name` and a `Pattern`. Where a format matches, the `MergeMessage.FormatName` property is populated from the `MergeMessageFormat.Name` property.

I also spotted that the `ParseVersion` function was receiving an irrelevant parameter, so fixed that.